File: backend/src/modules/rtpengine/rtpengine.service.ts
import { Injectable } from '@nestjs/common';
import { KamailioRpcService } from '../../common/services/kamailio-rpc.service';

@Injectable()
export class RtpengineService {
  constructor(private kamailioRpcService: KamailioRpcService) {}

  // RTPEngine 操作
  async showAll(): Promise<any> {
    return this.kamailioRpcService.call('rtpengine.show', ['all']);
  }

  async reload(): Promise<void> {
    await this.kamailioRpcService.call('rtpengine.reload');
  }

  async enable(url: string, flag: number): Promise<void> {
    await this.kamailioRpcService.call('rtpengine.enable', [url, flag]);
  }

  async ping(url: string): Promise<any> {
    return this.kamailioRpcService.call('rtpengine.ping', [url]);
  }

  // RTPProxy 操作
  async rtpproxyEnable(url: string, flag: number): Promise<void> {
    await this.kamailioRpcService.call('rtpproxy.enable', [url, flag]);
  }

  // NAT Helper
  async enableNatPing(flag: number): Promise<void> {
    await this.kamailioRpcService.call('nathelper.enable_ping', [flag]);
  }

  // 获取综合状态
  async getStatus(): Promise<any> {
    try {
      const rtpengine = await this.showAll();
      return { rtpengine, status: 'ok' };
    } catch (error) {
      return { status: 'error', error: error.message };
    }
  }
}

File: kam-admin-module-trace/src/main/java/io/kamailio/admin/modules/rtpengine/service/RtpengineService.java
package io.kamailio.admin.modules.rtpengine.service;

import io.kamailio.admin.common.RpcTimeouts;
import io.kamailio.admin.common.service.KamailioRpcService;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Map;

@Service
public class RtpengineService {
    private final KamailioRpcService kamailioRpc;

    public RtpengineService(KamailioRpcService kamailioRpc) {
        this.kamailioRpc = kamailioRpc;
    }

    public Object showAll() {
        return kamailioRpc.rtpengineShow("all").block(RpcTimeouts.DEFAULT_BLOCK);
    }

    public void reload() {
        kamailioRpc.rtpengineReload().block(RpcTimeouts.DEFAULT_BLOCK);
    }

    public void enable(String url, int flag) {
        kamailioRpc.rtpengineEnable(url, flag).block(RpcTimeouts.DEFAULT_BLOCK);
    }

    public Object ping(String url) {
        return kamailioRpc.rtpenginePing(url).block(RpcTimeouts.DEFAULT_BLOCK);
    }

    public void enableNatPing(int flag) {
        kamailioRpc.nathelperEnablePing(flag).block(RpcTimeouts.DEFAULT_BLOCK);
    }

    public Map<String, Object> getStatus() {
        try {
            Object rtpengine = showAll();
            Map<String, Object> result = new HashMap<>();
            result.put("rtpengine", rtpengine);
            result.put("status", "ok");
            return result;
        } catch (Exception e) {
            Map<String, Object> result = new HashMap<>();
            result.put("status", "error");
            result.put("error", e.getMessage());
            return result;
        }
    }
}
